### ISSUE TITLE
Significant fps drop to < 1/4 speed, generally laggy in gas-filled cave

### ISSUE THREAD

**Issue by matthewpost** [Regalis11/Barotrauma#6508]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
The game lags significantly, down to less than a quarter framerate in a particular mudraptor cave filled with gas.

**Steps To Reproduce**
- Find the attached save file [BaroSave_LaggyMudraptorCave.zip](https://github.com/Regalis11/Barotrauma/files/7069305/BaroSave_LaggyMudraptorCave.zip)
- Proceed down to the mudraptor cave.
- After removing the obstruction and passing the nest, proceed to the far back left, near the minerals
- Experience a rapid increase in lag at back of cave (see screenshot 2)
- Leave cave and everything is happy again

**Version**
Windows Steam release, 0.14.9.0, vanilla, no mods, just a tiny Dugong.

**Additional information**
100% of time for me in this save file, only in the one cave. I assume it is related to the gas and lighting.
I'm including a screenshot of the location and my graphics settings in case it helps.
If this can't be reproduced, it's something wrong on my end.
![image](https://user-images.githubusercontent.com/23487461/131186092-eb0a5a40-e43e-4133-b91e-7d58948d5615.png)
![image](https://user-images.githubusercontent.com/23487461/131192404-da11447c-721b-4f11-9da4-10b3069e0be2.png)

**Comment by Regalis11** [comment]
I'm afraid I couldn't reproduce this: there wasn't any significant change in the FPS in the cave at my end. I'd guess it has something to do with the hardware, e.g. maybe your GPU is a bit weaker than mine.

In any case, I do suspect the gas particles as well, since there's a huge number of those in that particular spot. I optimized the particle emitters a bit in https://github.com/Regalis11/Barotrauma-development/commit/cd35dc09cbc586073aeda612cb68b76a775c118a to hopefully improve the performance.

**Comment by matthewpost** [comment]
Thanks for the investigation! I figured this would be difficult to reproduce, and highly hardware-dependent. This issue was produced using a laptop with a GTX970M (yes, relatively old but runs the game comfortably). This was a rare occurrence for me. I suspect if the only thing unusual about that section of the cave is the gas, that's it.

I should have done a comparison between 100% particle limit and lower (or lighting resolution). I'll try reproducing again, and I'll see if switching the particle limit fixes it. If it does, you've already solved the problem.

**Comment by matthewpost** [comment]
I returned to the same cave, reproduced the issue and tested the graphic settings:
Default particle limit 100%, lighting 100% resolution, default LoS/fisheye/chromatic aberration ON
- Lag produced in same location
- Reducing particle limit to 0 instantly fixed the lagging issues
    - 10% particle limit is perfectly smooth
    - 20% particle limit has minor jitters in places
    - 30% unpleasant FPS drop
- Changing lighting resolution had no noticeable effect
- Changing Line of Sight changed nothing
- Disabling fisheye & chromatic aberration seemed to have no effect (I think it should have done something)

I'm including another screenshot of the location where I completed this comparative test. I'm running Steam release (vanilla) v0.14.9.0.

In conclusion, you were absolutely correct that it was the particle effects, and the other lighting/shader effects didn't have much of an impact. This also give anyone an easy workaround until the fix is released. Mission accomplished! Thanks!

![image](https://user-images.githubusercontent.com/23487461/131224474-6c3d0c54-bea5-4225-9fb0-ae00a81a371b.png)
